includes/class-wc-payments-order-service.php

```diff
@@ -161,10 +161,12 @@ public function mark_payment_started( $order, $intent_id, $intent_status, $charg
 	 * @return void
 	 */
 	public function mark_payment_capture_completed( $order, $intent_id, $intent_status, $charge_id ) {
-		if ( ! $this->order_prepared_for_processing( $order, $intent_id ) ) {
+		if ( ! is_a( $order, 'WC_Order' ) || $this->is_order_locked( $order, $intent_id ) ) {
```

Comment:
@dpaun1985, did we consider using `order_prepared_for_processing` here?

Reply:
that's what avoided, since that method check also the order status ( processing/completed ) . We can add a new param to `order_prepared_for_processing` though, so we can skip the order status check.

Reply:
Oh, I missed the original line :facepalm: .

Reply:
Looking into the codebase and workflows, I found `wc_get_is_paid_statuses` function in WooCommerce core.

The function name states what WC considers paid statuses. As the ticket reproduction steps have steps to set one of those statuses, it's a custom workflow.

So either this should be changed in WooCommerce core, or the merchant has to use `woocommerce_order_is_paid_statuses` hook to customize the paid statuses for their store.